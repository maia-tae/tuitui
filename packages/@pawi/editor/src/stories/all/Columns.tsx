// import * as React from 'react'
// import styled from 'styled-components'
import { Editor as component, EditorProps as Props } from '../..'
import { config, Stories, wrapper } from '../helpers'

/*
const SomeChild = styled.div`
  border: 1px solid #333;
  background: #aaa;
  height: 80px;
  padding: 20px;
  color: #eee;
`
*/

export const editorStories: Stories<Props> = {
  name: 'Editor',
  component,
  config,
  wrapper,
  stories: [
    {
      name: 'many pages',
      props(ctx) {
        return { holder: ctx.state.test.c1 }
      },
    },
  ],
}
