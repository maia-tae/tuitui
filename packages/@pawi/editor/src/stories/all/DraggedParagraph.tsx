import { reference } from '@pawi/build'
import styled from 'styled-components'
import {
  DraggedParagraph as component,
  DraggedParagraphProps as Props,
} from '../..'
import { config, Stories } from '../helpers'

const wrapper = styled.div`
  padding: 8px;
  background: white;
`

export const draggedParagraph: Stories<Props> = {
  name: 'Dragged paragraph',
  component,
  config,
  wrapper,
  stories: [
    {
      name: 'Long text',
      props(ctx) {
        const holder = ctx.state.test.e4
        const ids = Object.keys(holder.composition.g)
        const sourceId = ids[ids.length - 1]
        return {
          sourceRef: reference(holder),
          sourceId,
          sourceCompId: 'foo',
        }
      },
    },

    {
      name: 'Custom',
      props(ctx) {
        return {
          sourceRef: reference(ctx.state.test.e1),
          sourceId: 'bar',
          sourceCompId: 'foo',
        }
      },
    },
  ],
}
