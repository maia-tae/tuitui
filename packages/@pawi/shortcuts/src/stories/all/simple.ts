import {
  ShowMessage as component,
  ShowMessageProps as Props,
} from '../helpers/ShowMessage'
import { config, Stories } from '../helpers'

export const simpleStories: Stories<Props> = {
  name: 'Show shorcuts',
  component,
  config,
  stories: [
    {
      name: 'use arrows inside and outside input',
    },
  ],
}
