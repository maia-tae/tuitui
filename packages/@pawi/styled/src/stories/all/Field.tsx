import { Field as component, FieldProps as Props } from '../..'
import { config, Stories } from '../helpers'

export const fieldStories: Stories<Props> = {
  name: 'Field',
  component,
  config,
  stories: [
    {
      name: 'simple',
      props: ({ state }) => ({ form: state.login, name: 'username' }),
      state: {
        login: {
          username: 'someuser',
        },
      },
    },

    {
      name: 'checkbox',
      props: ({ state }) => ({
        form: state.login,
        name: 'admin',
        type: 'checkbox',
        label: true,
      }),
      state: {
        login: {
          username: 'someuser',
        },
      },
    },

    {
      name: 'select',
      props: ({ state }) => ({
        form: state.login,
        name: 'collId',
        type: 'select',
        icon: 'user',
        label: true,
        includeBlank: true,
        options: [
          { key: 'one', value: 'This is one' },
          { key: 'two', value: 'This is two' },
        ],
      }),
      state: {
        login: {
          username: 'someuser',
        },
      },
    },

    {
      name: 'submit',
      props: app => ({
        form: app.state.login,
        name: 'username',
        submit: app.actions.login.submit,
      }),
      state: {
        login: {
          username: 'someuser',
        },
      },
    },

    {
      name: 'field icon',
      props: ({ state }) => ({
        form: state.login,
        name: 'username',
        icon: 'user',
      }),
      state: {
        login: {
          username: 'someuser',
        },
      },
    },

    {
      name: 'field icon true',
      props: ({ state }) => ({
        form: state.login,
        name: 'username',
        icon: true,
      }),
      state: {
        login: {
          password: '',
        },
      },
    },

    {
      name: 'field disabled reset',
      props: ({ state }) => ({
        form: state.login,
        name: 'username',
        disabled: true,
        reset: true,
        resetIcon: 'username',
        icon: true,
      }),
      state: {
        login: {
          username: 'Hello',
        },
      },
    },

    {
      name: 'huge fieldBorderRadius',
      props: ({ state }) => ({
        form: state.login,
        name: 'username',
        icon: 'user',
      }),
      state: {
        login: {
          username: 'John Lily',
        },
      },
      theme: {
        fieldBorderRadius: '15px',
      },
    },
  ],
}
