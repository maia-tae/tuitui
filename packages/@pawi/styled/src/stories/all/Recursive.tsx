import * as React from 'react'
import { Recursive, RecursiveProps as Props } from '../..'
import { Comp, styled } from '../../app'
import { config, Stories } from '../helpers'

const Wrapper = styled.div`
  border: 1px solid orange;
  padding: 3px 8px;
`

const View: Comp<{ item: { title: string } }> = ({ item }) => {
  return <Wrapper>{item.title}</Wrapper>
}

const component = styled(Recursive)`
  width: 15rem;
  & .Sub {
    padding-top: 2px;
  }
`

export const recursiveStories: Stories<Props> = {
  name: 'Recursive',
  component,
  config,
  stories: [
    {
      name: 'simple',
      props: {
        item: { title: 'Foo' },
        getOpen(ctx, item) {
          return ctx.state.open[item.title]
        },
        setOpen(ctx, item, open) {
          ctx.actions.setOpen({ title: item.title, open })
        },
        getChildren(ctx, item) {
          if (item.title === 'Foo') {
            return ['Bar', 'Baz'].map(title => ({ title }))
          } else if (item.title == 'Baz') {
            return ['Babar', 'Papa'].map(title => ({ title }))
          }
          return undefined
        },
        View,
      },
      actions: {
        setOpen(ctx: any, arg: { title: string; open: boolean }) {
          ctx.state.open[arg.title] = arg.open
        },
      },
      state: {
        open: {
          Foo: true,
        },
      },
    },
  ],
}
