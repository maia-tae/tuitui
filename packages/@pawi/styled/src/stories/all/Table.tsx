import * as React from 'react'

import { Table as component, TableRow } from '../..'
import { config, Stories } from '../helpers'
import { styled, Comp } from '../../app'
import { BottomTable } from '../../components'

const wrapper = styled.div`
  height: 200px;
  width: 200px;
`

export const tableStories: Stories<{}> = {
  component,
  config,
  wrapper,
  name: 'Table',
  stories: [
    {
      name: 'simple',
      children: Array.from({ length: 3 })
        .map((_, idx) => idx)
        .map(idx => (
          <TableRow key={idx}>
            Element
            {idx}
          </TableRow>
        )),
    },

    {
      name: 'BottomTable',
      component: BottomTable,
      children: Array.from({ length: 100 })
        .map((_, idx) => idx)
        .map(idx => (
          <TableRow key={idx}>
            Element
            {idx}
          </TableRow>
        )),
    },
  ],
}
