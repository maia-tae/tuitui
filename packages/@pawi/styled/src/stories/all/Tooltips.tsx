import { Tooltips as component, TooltipsProps as Props } from '../..'
import { config, Stories } from '../helpers'

export const tooltipsStories: Stories<Props> = {
  name: 'Tooltips',
  component,
  config,
  stories: [
    {
      name: 'tooltips',
    },
  ],
}
