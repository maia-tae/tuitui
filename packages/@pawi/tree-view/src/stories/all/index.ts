export * from './Tree-dragdrop'
export * from './Library'
export * from './Node'
export * from './DropTarget'
export * from './Tree'
