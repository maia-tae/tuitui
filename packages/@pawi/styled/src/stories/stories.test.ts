import { testStories } from '@pawi/story'
import * as stories from './all'

testStories({}, stories)
