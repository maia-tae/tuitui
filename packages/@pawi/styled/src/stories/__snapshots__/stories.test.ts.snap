// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon FontAwesome should render hasOn 1`] = `
<div
  className="theme-Icon-FontAwesome-hasOn"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          hasOn
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="folder" hasOn ...
          <div
            className="jsx"
          >
            &lt;IconComp icon="folder" hasOn highlighted={false} onClick={function onClick} /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="hasClick Icon sc-iwsKbI bSnVXb"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-folder fa-w-16 "
            data-icon="folder"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Icon FontAwesome should render hasOn is on 1`] = `
<div
  className="theme-Icon-FontAwesome-hasOn-is-on"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          hasOn is on
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="folder" hasOn ...
          <div
            className="jsx"
          >
            &lt;IconComp icon="folder" hasOn highlighted onClick={function onClick} /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="highlighted hasClick Icon sc-iwsKbI bSnVXb"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-folder-open fa-w-18 "
            data-icon="folder-open"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Icon FontAwesome should render large 1`] = `
<div
  className="theme-Icon-FontAwesome-large"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          large
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="user" large /&gt;
          <div
            className="jsx"
          >
            &lt;IconComp icon="user" large /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="large Icon sc-iwsKbI bSnVXb"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user fa-w-14 "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Icon FontAwesome should render medium 1`] = `
<div
  className="theme-Icon-FontAwesome-medium"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          medium
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="user" medium /...
          <div
            className="jsx"
          >
            &lt;IconComp icon="user" medium /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="medium Icon sc-iwsKbI bSnVXb"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user fa-w-14 "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Icon FontAwesome should render simple 1`] = `
<div
  className="theme-Icon-FontAwesome-simple"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          simple
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="user" /&gt;
          <div
            className="jsx"
          >
            &lt;IconComp icon="user" /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="Icon sc-iwsKbI bSnVXb"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user fa-w-14 "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Icon FontAwesome should render tip 1`] = `
<div
  className="theme-Icon-FontAwesome-tip"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          tip
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;IconComp icon="user" tip="use...
          <div
            className="jsx"
          >
            &lt;IconComp icon="user" tip="userTip" /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="Icon sc-iwsKbI bSnVXb"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user fa-w-14 "
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tooltips should render tooltips 1`] = `
<div
  className="theme-Tooltips-tooltips"
>
  <div
    className="sc-bxivhb kGrjIq"
  >
    <div
      className="sc-ifAKCX BXQJS"
    >
      <div
        className="sc-EHOje fpqoIO"
      >
        <div
          className="sc-bwzfXH dvulrY Title"
          style={
            Object {
              "cursor": "inherit",
            }
          }
        >
          tooltips
        </div>
        <div
          className="sc-bdVaJa gdMgPe"
        >
          &lt;Component /&gt;
          <div
            className="jsx"
          >
            &lt;Component /&gt;
          </div>
        </div>
      </div>
      <div
        className="sc-htpNat eIswBe"
      >
        <span
          className="sc-frDJqD grUgAy"
        >
          <span
            className="highlighted hasClick Icon sc-iwsKbI sc-hmzhuo jAaDgF"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-question-circle fa-w-16 "
              data-icon="question-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
